nixos/modules/hardware/all-firmware.nix

```diff
@@ -1,8 +1,9 @@
 { config, lib, pkgs, ... }:
 
-with lib;
-
 let
+  inherit (lib)
+    literalExpression mdDoc mkIf mkMerge mkOption mkRenamedOptionModule types;
```

Comment:
IMO this is not an improvement over just prefixing things with ``lib.`` or using ``with lib`` more locally.

Reply:
At this first moment I am more interested in make the expressions compile. More cosmetic modifications can be done later.

Reply:
I am talking about the general idea of your change. If the next person (for example me) is going to revert this again because maintaining a list of used functions is not worth it, then we didn't achieve anything.

Reply:
The "idea" of my change is just getting rid of `with`. 

For this draft phase, I believe it is easier to use `inherit (lib) every thing that was called;`. It will help a future automation.

I have no problem in prefixing `lib.` This can be done with the help of an Emacs quick&dirty magic keystroke. Indeed I will do this tonight.

Reply:
> And it is depressing to look at `meta = with pkgs.lib.maintainers; { maintainers = []; };`. Such a heavy machinery to empty lists?

It is empty to convey that the package needs a maintainer.

Reply:
> It is empty to convey that the package needs a maintainer.

What is the difference for `meta.maintainers = [ ];`?